### DISCUSSION TITLE
Buildroot

### DISCUSSION THREAD

**Discussion by peterfealey** [savonet/liquidsoap#2889] [Installation]
I'm relatively new to embedded Linux and am working on a project using Buildroot. 

Has anyone successfully installed Liquidsoap to an image using Buildroot, and if so would they mind sharing a little about how they achieved it?

Thanks!

**Comment by toots** [comment]
Hi! I never tried but it would be nice!

Two options:
1. Native build (even with a qemu/emulated env)
2. Cross-compilation build

Option 1 would be easier even if it takes longer. Cross-compilation is definitely possible (we do it on windows) but it requires a _lot_ of intermediary packages to be prepared for it.

A couple more notes:
* Do you know the external/optional dependencies you would like to include in liquidsoap? The less dependencies the easier your build process will be.
* Liquidsoap being built using OCaml, it takes advantage of its memory management. This means that, for an embedded system, you might not have much control over when and how much memory is allocated. However, OCaml is generally pretty good at it and we also expose all the settings from OCaml to change how the memory management behaves. Here too, the less dependencies you have the better you would be.

**Comment by peterfealey** [comment]
Hi, thanks for getting back to me. 

I think the following would certainly cover it for me.

**For sure:**
ffmpeg
alsa
cry
lame

**Maybe:**
flac
ogg
opus

Background is I'm using Darkice at the moment with a local and remote stream. If the remote server is down, Darkice crashes out for even the local, which is a non-starter for me. 

In terms of the memory management, my board has 1GB with not a lot used elsewhere. It's a RockPi S if you're interested :)

Thanks!

**Comment by peterfealey** [comment]
@toots - did you have any guidance on this? I know it's probably a bit of an edge-case.
